common/src/main/java/com/wynntils/features/user/overlays/InfoBoxesFeature.java

```diff
@@ -0,0 +1,218 @@
+
+    public static class InfoBoxOverlay extends Overlay {
+        @Config
+        public FontRenderer.TextShadow textShadow = FontRenderer.TextShadow.OUTLINE;
+
+        @Config
+        public String content = "";
+
+        private final int id;
+
+        private TextRenderTask toRender = recalculateFunctions(content);
+        private List<Function<?>> functionDependencies = new ArrayList<>();
+
+        public InfoBoxOverlay(int id) {
+            super(
+                    new OverlayPosition(
+                            -60 + (15 * id),
+                            5,
+                            VerticalAlignment.Top,
+                            HorizontalAlignment.Left,
+                            OverlayPosition.AnchorSection.MiddleLeft),
+                    new GuiScaledOverlaySize(120, 10),
+                    HorizontalAlignment.Left,
+                    VerticalAlignment.Middle);
+            this.id = id;
+        }
+
+        private TextRenderTask recalculateFunctions(String renderableText) {
+            if (functionDependencies == null) {
+                functionDependencies = new ArrayList<>();
+            }
+
+            for (Function<?> oldDependency : functionDependencies) {
+                FunctionManager.disableFunction(oldDependency);
+            }
+
+            functionDependencies.clear();
+
+            StringBuilder builder = new StringBuilder(renderableText.length() + 10);
+            Matcher m = INFO_VARIABLE_PATTERN.matcher(renderableText);
+            while (m.find()) {
+                String replacement = null;
+                if (m.group(1) != null && FunctionManager.forName(m.group(1)).isPresent()) {
+                    // %variable%
+                    Function<?> function = FunctionManager.forName(m.group(1)).get();
+
+                    FunctionManager.enableFunction(function);
+                    functionDependencies.add(function);
+
+                    replacement = FunctionManager.getRawValueString(function, "");
+                } else if (m.group(2) != null) {
+                    // \escape
+                    replacement = doEscapeFormat(m.group(2));
+                }
+                if (replacement == null) {
+                    replacement = m.group(0);
+                }
+
+                m.appendReplacement(builder, replacement);
+            }
+            m.appendTail(builder);
+
+            return new TextRenderTask(
+                    parseColorCodes(builder.toString()),
+                    TextRenderSetting.DEFAULT
+                            .withHorizontalAlignment(this.getRenderHorizontalAlignment())
+                            .withMaxWidth(this.getWidth())
+                            .withCustomColor(CustomColor.fromChatFormatting(ChatFormatting.WHITE))
+                            .withTextShadow(textShadow));
+        }
+
+        private String parseColorCodes(String toProcess) {
```

Comment:
This too, if kept, should move out of the Feature itself. Either to FunctionManager (if we think this is strictly something we will do for function templates), or to like StringUtils or ComponentUtils (which include support for coded strings as well)